### PR TITLE
[LLM Runtime] Support load_in_nbit in llm runtime

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

Support load_in_nbit in llm runtime

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/transformers/modeling/modeling_auto.py [hunk 1]
```diff
@@ -101,9 +101,14 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
                 torch_dtype = kwargs.pop("torch_dtype", torch.float32)
             if load_in_4bit:
                 if quantization_config is None:
-                    quantization_config = WeightOnlyQuantConfig(
-                        compute_dtype=torch_dtype, weight_dtype="nf4"
-                    )
+                    if use_llm_runtime: 
+                        quantization_config = WeightOnlyQuantConfig(
+                            compute_dtype="int8", weight_dtype="int4"
```
Comment (hshen14) [inline intel_extension_for_transformers/transformers/modeling/modeling_auto.py:106]: compute_type should be bf16 also

**Comment by kevinintel** [comment]
please add new API in main page and graph readme

File: intel_extension_for_transformers/transformers/modeling/modeling_auto.py (hunk 1 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/transformers/modeling/modeling_auto.py:106]: compute_type=bf16 can only run on spr machine

**Comment by zhenwei-intel** [comment]
> please add new API in main page and graph readme

updated

File: README.md [hunk 2]
```diff
@@ -61,32 +61,30 @@ Below is the sample code to enable weight-only INT4/INT8 inference. See more [ex
 ### INT4 Inference 
 ```python
 from transformers import AutoTokenizer, TextStreamer
-from intel_extension_for_transformers.transformers import AutoModelForCausalLM, WeightOnlyQuantConfig
+from intel_extension_for_transformers.transformers import AutoModelForCausalLM
 model_name = "Intel/neural-chat-7b-v1-1"     # Hugging Face model_id or local model
-config = WeightOnlyQuantConfig(compute_dtype="int8", weight_dtype="int4")
```
Comment (a32543254) [inline README.md:66]: should we also give the example of WeightOnlyQuantConfig for advance usage ? 
@hshen14

**Review by a32543254 — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 3]
```diff
@@ -73,7 +73,7 @@ tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
 inputs = tokenizer(prompt, return_tensors="pt").input_ids
 streamer = TextStreamer(tokenizer)
 
-model = AutoModelForCausalLM.from_pretrained(model_name, quantization_config=config)
+model = AutoModelForCausalLM.from_pretrained(model_name, load_in_4bit=True)
```
Comment (DDEle) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:76]: Remove prior `config` definition if not used.

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:75]: thanks, updated

**Review by hshen14 — APPROVED** [review]

**Comment by VincyZhang** [comment]
This PR lead to Neuralchat llm_runtime_int4_server hangs, would you please figure out the reason? @zhenwei-intel  @lvliang-intel

**Pull request merged**